### PR TITLE
feat: expose bn254 commitment computation with G1 elements (PROOF-789)

### PR DESCRIPTION
**Pull request by jacobtrombetta**
# Rationale for this change
The `blitzar-sys` crate has the ability to compute commitments with `bn254` `G1` elements. This work exposes that capability from the `blitzar` crate.

Follow up work will rename the functions in the API. The name is chosen to align with our current function naming practice.

# What changes are included in this PR?
- Added `compute_bn254_g1_uncompressed_commitments_with_generators` and tests.
- Added documentation to support `bn254` commitment computation.
- Updated the commands to run examples in the examples document.
- Documentation is updated to show we now support `bn254` commitment computation with `G1` elements.

# Are these changes tested?
- Yes

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -5,7 +5,7 @@ name = "blitzar"
 edition = "2021"
 license = "Apache-2.0"
 repository = "https://github.com/spaceandtimelabs/blitzar-rs"
-keywords = ["gpu-cryptography", "gpu-crypto", "gpu-ristretto", "gpu-curve25519", "gpu-ristretto255"]
+keywords = ["gpu-cryptography", "gpu-crypto", "gpu-ristretto", "gpu-curve25519", "gpu-ristretto255", "gpu-bls12-381", "gpu-bn254"]
```
Comment (JayWhite2357) [inline Cargo.toml]: I'm not sure on naming conventions here, but perhaps things like `bls12-381` and `bn254` are better/additional keywords?

File: src/compute/commitments.rs [hunk 2]
```diff
@@ -139,6 +140,44 @@ pub fn compute_bls12_381_g1_commitments_with_generators(
...
+/// # Example - Pass generators to Commitment Computation
+///```no_run
+#[doc = include_str!("../../examples/pass_bn254_g1_generators_to_commitment.rs")]
+///```
+pub fn compute_bn254_g1_uncompressed_commitments_with_generators(
+    commitments: &mut [bn254_g1_affine],
+    data: &[Sequence],
+    generators: &[bn254_g1_affine],
+) {
+    init_backend();
```
Comment (JayWhite2357) [inline src/compute/commitments.rs:154]: Should this be in here? Is it free if it's already been initialized?

**Review by JayWhite2357 — DISMISSED** [review]

File: Cargo.toml (hunk 1 above)
Comment (jacobtrombetta) [reply Cargo.toml]: I'm not sure the motivation behind the tag naming and can't find any guidance online about naming conventions for keywords.

Perhaps the `gpu-*` was added to help with searching? For example searching `curve25519` on crates.io produces 165 results. `gpu-curve25519` produces two results - `blitzar` and `blitzar-rs`. On the flip side searching `curve25519-gpu` or `curve25519 gpu` produce zero results.

It seems like we should drop `gpu-*` from all of the curve names, including `gpu-ristretto`, `gpu-curve25519`, and `gput-ristretto255` as well? I'm thinking we should make it:

`keywords = ["gpu", "cryptography", "crypto", "ristretto", "curve25519", "ristretto255", "bls12-381", "bn254"]`

What do you think?

File: src/compute/commitments.rs (hunk 2 above)
Comment (jacobtrombetta) [reply src/compute/commitments.rs:154]: Great question. The `init_backend()` method call here just ensures the backend was in fact called. See [here](https://github.com/spaceandtimelabs/blitzar-rs/blob/e1ff8e0dcf79b87fd6b61969433361b71f0270c5/src/compute/backend.rs#L50-L70). If it has already been called, it simply returns without doing anything.

The other `compute_*_commitment` methods have the same call to `init_backend()`.

File: Cargo.toml (hunk 1 above)
Comment (JayWhite2357) [reply Cargo.toml]: Sounds good to me.

**Review by JayWhite2357 — APPROVED** [review]

**Pull request merged**